src/github.com/matrix-org/go-neb/services/imgur/imgur.go

```diff
@@ -0,0 +1,268 @@
+	Views       int64  `json:"views"`       // The number of image views
+	Link        string `json:"link"`        // The direct link to the the image. (Note: if fetching an animated GIF that was over 20MB in original size, a .gif thumbnail will be returned)
+	Gifv        string `json:"gifv"`        // OPTIONAL, The .gifv link. Only available if the image is animated and type is 'image/gif'.
+	MP4         string `json:"mp4"`         // OPTIONAL, The direct link to the .mp4. Only available if the image is animated and type is 'image/gif'.
+	MP4Size     int64  `json:"mp4_size"`    // OPTIONAL, The Content-Length of the .mp4. Only available if the image is animated and type is 'image/gif'. Note that a zero value (0) is possible if the video has not yet been generated
+	Looping     bool   `json:"looping"`     // OPTIONAL, Whether the image has a looping animation. Only available if the image is animated and type is 'image/gif'.
+	NSFW        bool   `json:"nsfw"`        // Indicates if the image has been marked as nsfw or not. Defaults to null if information is not available.
+	Topic       string `json:"topic"`       // Topic of the gallery image.
+	Section     string `json:"section"`     // If the image has been categorized by our backend then this will contain the section the image belongs in. (funny, cats, adviceanimals, wtf, etc)
+	IsAlbum     bool   `json:"is_album"`    // If it's an album or not
+	// ** Unimplemented fields **
+	// bandwidth	integer	Bandwidth consumed by the image in bytes
+	// deletehash	string	OPTIONAL, the deletehash, if you're logged in as the image owner
+	// comment_count	int	Number of comments on the gallery image.
+	// topic_id	integer	Topic ID of the gallery image.
+	// vote	string	The current user's vote on the album. null if not signed in or if the user hasn't voted on it.
+	// favorite	boolean	Indicates if the current user favorited the image. Defaults to false if not signed in.
+	// account_url	string	The username of the account that uploaded it, or null.
+	// account_id	integer	The account ID of the account that uploaded it, or null.
+	// ups	integer	Upvotes for the image
+	// downs	integer	Number of downvotes for the image
+	// points	integer	Upvotes minus downvotes
+	// score	integer	Imgur popularity score
+}
+
+type imgurGalleryAlbum struct {
+	ID          string              `json:"id"`           // The ID for the album
+	Title       string              `json:"title"`        // The title of the album.
+	Description string              `json:"description"`  // Description of the album.
+	DateTime    int64               `json:"datetime"`     // Time inserted into the gallery, epoch time
+	Views       int64               `json:"views"`        // The number of album views
+	Link        string              `json:"link"`         // The URL link to the album
+	NSFW        bool                `json:"nsfw"`         // Indicates if the album has been marked as nsfw or not. Defaults to null if information is not available.
```

Comment:
Not that we actually do anything with this, but for tri-state flags like this (true/false/null) you'll need to use `*bool` otherwise `null` will be treated as `false`. https://play.golang.org/p/US6V2qiykX